src/Language/PureScript/Types.hs

```diff
@@ -109,12 +107,30 @@ data Type
   | ParensInType Type
   deriving (Show, Read, Eq, Ord)
 
--- |
--- A typeclass constraint
---
-type Constraint = (Qualified (ProperName 'ClassName), [Type])
+-- | Additional data relevant to type class constraints
+data ConstraintData
+  = PartialConstraintData [[String]] Bool -- ^ a constraint generated by the exhaustivity checker
```

Comment:
Do you think you could perhaps make the comments around this data declaration a little clearer? What do the arguments mean, for instance?